runtime/vm/ValueTypeHelpers.hpp

```diff
@@ -119,7 +119,7 @@ class VM_ValueTypeHelpers {
 					U_64 lhsValue = objectAccessBarrier.inlineMixedObjectReadU64(currentThread, lhs, startOffset + result->offset);
 					U_64 rhsValue = objectAccessBarrier.inlineMixedObjectReadU64(currentThread, rhs, startOffset + result->offset);
 
-					if (!checkDoubleEquality(lhsValue, rhsValue)) {
```

Comment:
i think we can just do u32 read for b,s,c,i,f and u64 for j,d

Reply:
Updated.